test-case/check-sof-logger.sh

```diff
@@ -48,23 +48,29 @@ fi
 dlogi "Found file: $(md5sum $ldcFile|awk '{print $2, $1;}')"
 
 tmp_file=/tmp/$RANDOM.sof-logger.log
+err_tmp_file=/tmp/err_logger.log
 
 func_lib_check_sudo
 
 dlogi "Try to dump the dma trace log via sof-logger ..."
-dlogc "sudo $loggerBin -t -l $ldcFile -o $tmp_file 2>&1 &"
-sudo $loggerBin -t -l $ldcFile -o $tmp_file 2>&1 &
+# sof-logger errors will output to $err_tmp_file
+dlogc "sudo $loggerBin -t -l $ldcFile -o $tmp_file 2> $err_tmp_file &"
+sudo $loggerBin -t -l $ldcFile -o $tmp_file 2> $err_tmp_file &
 sleep 2
 dlogc "sudo pkill -9 $(basename $loggerBin)"
 sudo pkill -9 $(basename $loggerBin) 2> /dev/null
 
-# get size of trace log$
-size=`du -k $tmp_file | awk '{print $1}'`
-if [[ $size -lt 1 ]]; then
-    dloge "No available log export."
+# check if we get any sof-logger errors
+logger_err=`grep -E "Error|error|ERROR" $err_tmp_file`
+if [[ $logger_err ]]; then
+    dloge "No available log to export due to sof-logger errors."
+    sudo rm -rf $tmp_file $err_tmp_file
     exit 1
 fi
 dlogi "Log data BEG>>"
 cat $tmp_file
 dlogi "<<END log data"
+
+sudo rm -rf $tmp_file $err_tmp_file
```

Comment:
Do we need to clean up the tmp files? I think keep them for later debug is better.